server/sonar-server/src/main/java/org/sonar/server/settings/ws/SetAction.java

```diff
@@ -229,4 +343,18 @@ private PropertyDto toProperty(SetRequest request, Optional<ComponentDto> compon
 
     return property;
   }
+
+  private static PropertyDto toFieldProperty(KeyValue keyValue, @Nullable Long componentId) {
+    return new PropertyDto().setKey(keyValue.key).setValue(keyValue.value).setResourceId(componentId);
+  }
+
+  private static class KeyValue {
```

Comment:
In my opinion, you should add more logic here : 
- The constructor should take the baseSettingKey, the fieldId, the Entry and the component
- The toFieldProperty could be moved inside